core/grants.go

```diff
@@ -47,7 +47,7 @@ func (a *API) createGrant(ctx context.Context, x apiGrant) error {
 	// so we should do our equality comparisons accordingly.
 	guardData, err := json.Marshal(x.GuardData)
 	if err != nil {
-		return errors.Wrap(err)
+		return nil, errors.Wrap(err)
 	}
 
 	g := authz.Grant{
```

Comment:
Why not just return `g` if `err == nil` at the end of this, instead of modifying `StoreGrant`?